app/actions/remote/channel.ts

```diff
@@ -97,6 +98,81 @@ export const fetchChannelByName = async (serverUrl: string, teamId: string, chan
     }
 };
 
+export function generateChannelNameFromDisplayName(displayName: string) {
+    let name = cleanUpUrlable(displayName);
+    if (name === '') {
+        name = generateId();
+    }
+    return name;
+}
+
+export const handleCreateChannel = async (serverUrl: string, displayName: string, purpose: string, header: string, type: ChannelType) => {
+    const database = DatabaseManager.serverDatabases[serverUrl]?.database;
+    if (!database) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    let client: Client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    try {
+        const currentUserId = await queryCurrentUserId(database);
+        const currentTeamId = await queryCurrentTeamId(database);
+        const name = generateChannelNameFromDisplayName(displayName);
+        const channel = {
+            creator_id: currentUserId,
+            team_id: currentTeamId,
+            display_name: displayName,
+            header,
+            name,
+            purpose,
+            type,
+        } as Channel;
+
+        const channelData = await client.createChannel(channel);
+        if (channelData?.id) {
+
```

Comment:
Create a ticket for now.  Need to hook up these actions after the channel view with posts is available.

Reply:
I'll implement instead of creating the ticket.